cub/cub/agent/agent_merge_sort.cuh

```diff
@@ -508,6 +489,8 @@ struct AgentMerge
   template <bool IS_FULL_TILE>
   _CCCL_DEVICE _CCCL_FORCEINLINE void consume_tile(int tid, OffsetT tile_idx, OffsetT tile_base, int count)
   {
+    _CCCL_PDL_GRID_DEPENDENCY_SYNC();
+
     const OffsetT partition_beg = merge_partitions[tile_idx + 0];
     const OffsetT partition_end = merge_partitions[tile_idx + 1];
```

Comment:
This is the actual fix. We are reading from global memory here via `merge_partitions`, but the `_CCCL_PDL_GRID_DEPENDENCY_SYNC` came later. Because this kernel can overlap into the previous one it may read invalid data and later access shared memory out of bounds, leading to `cudaErrorIllegalAddress` reported.

Adding `_CCCL_PDL_TRIGGER_NEXT_LAUNCH();` in the previous kernel just seems to make this issue happen more likely, but it is not necessary for the race condition to occur.